app/src/main/java/com/team41/boromi/book/OwnedFragment.java

```diff
@@ -2,35 +2,34 @@
 
 import android.os.Bundle;
 
+import android.util.Pair;
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentManager;
 import androidx.fragment.app.FragmentTransaction;
 
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.Button;
 
+import androidx.viewpager2.widget.ViewPager2;
+import com.google.android.material.tabs.TabItem;
+import com.google.android.material.tabs.TabLayout;
+import com.google.android.material.tabs.TabLayout.OnTabSelectedListener;
+import com.google.android.material.tabs.TabLayout.Tab;
+import com.team41.boromi.adapters.PagerAdapter;
 import com.team41.boromi.R;
+import java.util.ArrayList;
 
 /**
  * A simple {@link Fragment} subclass. Use the {@link OwnedFragment#newInstance} factory method to
  * create an instance of this fragment.
  */
 public class OwnedFragment extends Fragment {
 
-  // TODO: Rename parameter arguments, choose names that match
-  // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-  private static final String ARG_PARAM1 = "param1";
-  private static final String ARG_PARAM2 = "param2";
-
-  // TODO: Rename and change types of parameters
-  private String mParam1;
-  private String mParam2;
-
-  private FragmentManager manager = null;
-  FragmentTransaction ft;
-  GenericListFragment activeFragment;
+  private TabLayout tabLayout;
+  private ViewPager2 viewPager2;
+  private PagerAdapter pagerAdapter;
+  private ArrayList<Pair<Class<? extends Fragment>, Bundle>> fragmentClasses = new ArrayList<Pair<Class<? extends Fragment>, Bundle>>();
```

Comment:
As discussed, I think we def eventually want to do something about these in the future. This is quite a lot of ArrayList objects

Reply:
I agree with changing bundle to something else in the future. Realistically, only PagerAdapter needs an instance of ArrayList. I'll update it soon